vignettes/basic-principles.Rmd

```diff
@@ -27,133 +42,93 @@ In practice, this is done by labeling each record in the data with three additio
 * from\_ts: the timestamp at which the record was introduced to the dataset
 * until\_ts: the timestamp at which the record was removed from the dataset
 
-This strategy of time versioning is often called "type 2" history.
+This strategy of time versioning is often called "type 2" history [@Kimball2013].
 
 Note that identical records may be removed and introduced more than once; for example, in a table of names and addresses, a person may change their address (or name) back to a previous value.
 
 The SCDB package provides the function `update_snapshot` to handle the insertion and deactivation of records using this strategy.
 It further includes several functions to improve the Quality of life for working with database data.
 
-A simple example is shown below:
+A simple example of usage is shown below.<br/>
+For this example, we use a temporary, on-disk SQLite database.
+Note that `get_connection` tries to establish connection using `DBI::dbConnect` with as few additional arguments as possible.
+Different drivers may require authentication which can be read from a configuration file.^[
+In the context of the SCDB package, this is most notably `RPostgres::Postgres()`, which may read from a `.pgpass` file.
+See also the [PostgreSQL documentation](https://www.postgresql.org/docs/current/libpq-pgpass.html).
+]
 
-```{r example_conn}
-# First we connect to our database.
-# If this is, e.g., a PostgreSQL database already running on the machine, connection
-# can be done after the configuration of a .pgpass file.
-# For this example, we use an on-disk SQLite db to showcase.
-conn <- get_connection(drv = RSQLite::SQLite())
-# NOTE: Had the PostgreSQL DB been configured, we would not need to pass any args
-# to get_connection()
-```
-
-```{r example_data}
-# Our example data is mtcars with rownames converted to a column and only the first hp
-# column of mtcars
-example_data <- dplyr::transmute(mtcars, car = rownames(mtcars), hp)
-# If the data does not already live on the remote, we must transfer it
-example_data <- dplyr::copy_to(conn, example_data, overwrite = TRUE)
+Our example data is `datasets::mtcars` reduced to only two columns: row names converted to a column `car`, and `hp`
+```{r example_data, eval = FALSE}
```

Comment:
This block does not print the example data, is that intended?
I think it would be more easy to follow if the content of `example_data` is printed to the readers.

Reply:
Oops - that was naturally not intended. I've addressed this now as well.